core/src/test/java/com/linecorp/armeria/server/docs/JsonSchemaGeneratorTest.java

```diff
@@ -0,0 +1,159 @@
+                             DescriptionInfo.of("param2 description")),
+                FieldInfo.of("param3", TypeSignature.ofBase("string"),
+                             DescriptionInfo.of("param3 description")),
+                FieldInfo.of("param4", TypeSignature.ofBase("boolean"),
+                             DescriptionInfo.of("param4 description")));
+        final StructInfo structInfo = newStructInfo(methodName, parameters);
+
+        final ServiceSpecification serviceSpecification = generateServiceSpecification(structInfo);
+        final JsonNode jsonSchema = JsonSchemaGenerator.generate(serviceSpecification).get(0);
+
+        // Base properties
+        assertThat(jsonSchema.get("title").asText()).isEqualTo(methodName);
+        assertThat(jsonSchema.get("description").asText()).isEqualTo(methodDescription.docString());
+        assertThat(jsonSchema.get("type").asText()).isEqualTo("object");
+
+        // Method specific properties
+        final List<Entry<String, JsonNode>> properties =
+                ImmutableList.copyOf(jsonSchema.get("properties").fields());
+        assertThat(properties).hasSize(4);
+
+        assertThat(jsonSchema.get("properties").get("param1").get("type").asText()).isEqualTo("integer");
+        assertThat(jsonSchema.get("properties").get("param2").get("type").asText()).isEqualTo("number");
+        assertThat(jsonSchema.get("properties").get("param3").get("type").asText()).isEqualTo("string");
+        assertThat(jsonSchema.get("properties").get("param4").get("type").asText()).isEqualTo("boolean");
+    }
+
+    @Test
+    void testMethodWithRecursivePath() {
+        final Object commonTypeObjectForRecursion = new Object();
+        final List<FieldInfo> parameters = ImmutableList.of(
+                FieldInfo.of("param1", TypeSignature.ofBase("int"), DescriptionInfo.of("param1 description")),
+                FieldInfo.builder("paramRecursive", TypeSignature.ofStruct("rec", commonTypeObjectForRecursion))
+                         .build()
+        );
+
+        final StructInfo structInfo = newStructInfo(methodName, parameters);
+
+        final List<FieldInfo> parametersOfRec = ImmutableList.of(
+                FieldInfo.of("inner-param1", TypeSignature.ofBase("int32")),
+                FieldInfo.of("inner-recurse", TypeSignature.ofStruct("rec", commonTypeObjectForRecursion))
+        );
+        final StructInfo rec = newStructInfo("rec", parametersOfRec);
+
+        final ServiceSpecification serviceSpecification = generateServiceSpecification(structInfo, rec);
+        final JsonNode jsonSchema = JsonSchemaGenerator.generate(serviceSpecification).get(0);
+
+        assertThat(jsonSchema.get("properties").isNull()).isFalse();
+        assertThat(jsonSchema.get("properties").get("paramRecursive")).isNotNull();
+        assertThat(jsonSchema.get("properties").get("paramRecursive").get("properties")
+                             .get("inner-param1")).isNotNull();
```

Comment:
I prefer `assertThatJson` for assertions on json since:

1. Less clutter checking for nullability using json paths
2. If an assertion fails, more useful information is printed.

Ditto for the other assertions

```
assertThatJson(jsonSchema).node("properties.paramRecursive.properties.inner-param1.type").isStringEqualTo("integer");
```

Reply:
Sorry 😅   Do you mind also applying `assertThat` -> `assertThatJson` for tests in this class? (like done in `GrpcDocServiceJsonSchemaTest`?